besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2122,7 +2141,7 @@ private PrivacyParameters privacyParameters(final KeyValueStorageProvider storag
       } else if (hasPrivacyPublicKey) {
         throw new ParameterException(
             commandLine, "Privacy multi-tenancy and privacy public key cannot be used together");
-      } else if (!isPrivacyMultiTenancyEnabled) {
+      } else if (!isPrivacyMultiTenancyEnabled && !isUnrestrictedPrivacyEnabled) {
```

Comment:
`unrestricted` privacy isn't encrypting anything, yet. I don't think there's a nice generic way, to implement unrestricted privacy with regards to payload encryption. As such I think we should delegate responsibility to the user via a plugin or not encrypt at all. We could use a symmetric key for encrypting the payload but I don't think it adds much value to the user.

Reply:
I've caught up - we need the privacyPublicKeyFile for privateFrom! Going to refactor that code a little to make it more readable.

Reply:
Yes, I guess you are right. I think that there are probably a number of possibilities how this could be handled. If you had the public keys of all the recipients you could use the same encryption method that is used in Tessera, or if you prefer a more general approach you could use CMS messages, using the tools that are available for PKI, like X509 certificates, CRLs, LDAP (to find the right keys to use), more flexibility with crypto algorithms ...
You could also have a single secret key for a privacy group, which would probably allow for better "privacy", and better performance as well. 
So many ways to do that ...
Do you have a plan on what you are going to do?

Reply:
https://github.com/hyperledger/besu/issues/2370